src/main/java/org/jabref/logic/util/io/CiteKeyBasedFileFinder.java

```diff
@@ -32,40 +33,34 @@
     }
 
     @Override
-    public Map<BibEntry, List<Path>> findAssociatedFiles(List<BibEntry> entries, List<Path> directories, List<String> extensions) {
+    public Multimap<BibEntry, Path> findAssociatedFiles(List<BibEntry> entries, List<Path> directories, List<String> extensions) {
         Objects.requireNonNull(directories);
         Objects.requireNonNull(entries);
 
-        Map<BibEntry, List<Path>> result = new HashMap<>();
+        Multimap<BibEntry, Path> result = ArrayListMultimap.create();
```

Comment:
If you structure the code similar to the one in AutoSetLinks, you won't need a multimap:
```
 for (BibEntry entry : entries) {

                List<LinkedFile> linkedFiles = util.findassociatedNotLinkedFiles(entry, databaseContext, Globals.prefs.getFileDirectoryPreferences(), Globals.prefs.getAutoLinkPreferences(), ExternalFileTypes.getInstance());

                if (ce != null) {
                    for (LinkedFile linkedFile : linkedFiles) {
```

Reply:
Thanks! This is really good feedback. I just realized that the finder never gets a list of entries as argument but always a single entry. Thus the design can be simplified drastically without mulitmaps or maps to collections.